pkg/containerruntimediscovery/containerddiscoverer.go

```diff
@@ -0,0 +1,202 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package containerruntimediscovery
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/containerd/containerd"
+	criConstants "github.com/containerd/containerd/pkg/cri/constants"
+	"github.com/containerd/nerdctl/pkg/imgutil"
+	"github.com/containerd/nerdctl/pkg/labels/k8slabels"
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openclarity/vmclarity/api/models"
+	"github.com/openclarity/vmclarity/pkg/shared/log"
+	"github.com/openclarity/vmclarity/pkg/shared/utils"
+)
+
+type ContainerdDiscoverer struct {
+	client *containerd.Client
+	logger *logrus.Entry
```

Comment:
I personally dislike the approach of storing the logger in any struct instead of retrieving the logger from context mainly because there is not way to provide request specific context to be logged if the initial logger is reused.

I think this is a bad practice so I'd prefer to remove Logger from struct and get it from context.Context whereever it is used.

Reply:
I think I caught all the new places in the code doing this and added the context where it was missing, please check.